File: examples/example_gi.lua
#!/usr/bin/env lua

-- example_gi.lua
--
-- Created in 2011 by Ulrik Sverdrup <ulrik.sverdrup@gmail.com>
-- This work is placed in the public domain.
--
-- requires lgi  http://gitorious.org/lgi

local lgi = require 'lgi'
local Gtk = lgi.require('Gtk', '2.0')
local Keybinder = lgi.require('Keybinder', '0.0')

local function callback(keystring)
  print("In callback for", keystring)
  print("Event time:", Keybinder.get_current_event_time())
  Keybinder.unbind(keystring)
  Gtk.main_quit()
end

Gtk.init()
Keybinder.init()
Keybinder.bind("<Control>A", callback)
print("Press <Control>A to activate keybinding and quit");
Gtk.main()

File: examples/example_gi.py
#!/usr/bin/python
"""
example_gi.py

Created in 2013 by Martin Kolman <martin.kolman@gmail.com>
This work is placed in the public domain.
"""
from gi.repository import Gtk
from gi.repository import Keybinder


def callback(key_string, user_data):
    """The callback gets the keystring pressed and any user provided data."""
    print "Pressed:", key_string
    print "Handling", user_data
    Gtk.main_quit()

if __name__ == '__main__':
    keystr = "<Ctrl>A"
    # the GI provided Keybinder needs
    # to be initialized before use
    Keybinder.init()
    Keybinder.bind(keystr, callback, "Keystring %s (user data)" % keystr)
    print "Press", keystr, "to handle keybinding and quit"
    Gtk.main()
